joplin/settings.py

```diff
@@ -146,9 +146,16 @@
 #
 default_db_url = f'sqlite:///{os.path.join(PROJECT_DIR, "db.sqlite3")}'
 DATABASES = {
-    'default': dj_database_url.config(default=default_db_url),
+    'default': dj_database_url.config(default=default_db_url, engine='django_postgrespool2', conn_max_age=0),
 }
 
+DATABASE_POOL_CLASS = 'sqlalchemy.pool.QueuePool'
+# https://github.com/lcd1232/django-postgrespool2#configuration
+DATABASE_POOL_ARGS = {
```

Comment:
there is a relationship between this and the gunicorn settings, but I won't claim to fully understand it. The gunicorn workers are spun up and managed through a combo of the OS thread managment and the application. 

The gist here is: heroku hobby tiers allow for 20 connections, so we're assuming heroku itself may consume some of those, so we don't ever want to go more than 10 connections, but we expect around 5. 

** NOTE ** I'm _not_ totally clear on whether this applies to _each_ gunicorn worker or all of them together. 
** ALSO NOTE ** Each heroku dyno seems to spin up its own instance, but I'm _not_ sure how that affects the number of potential connections on the database. This might have been where I was running into problems running out of connections even with conservative pooling settings. 

These pooling settings refer specifically to the application, so settings here will affect what gunicorn ends up having to handle (I think). 

The 'recycle' setting  is kinda interesting, more info on that in the SQL alchemy docs themselves.

Reply:
> This might have been where I was running into problems running out of connections even with conservative pooling settings.

Do we run out of connections if we have multiple dynos running? Under what conditions do we have problems running out of connections even with conservative pooling settings?

Reply:
We run out of connections if we change gunicorn to run asynchronously but don't implement any form of connection pooling. 

We run out of connections with pooling on a PR app if we set these pool size settings too high. If we set them lower, we don't run into any problems. 

We could set up some tests if we want to see where specifically we run into problems with connection pooling, but its not going to be a problem in staging or production with these settings. 

We could also implement connection pooling using pgbouncer, which I'd expect to have more predictable results.